### PR TITLE
APM-207673 CI Adoptions

### PR DESCRIPTION
**Pull request by thomschuetz**
This pr includes following:
- remove deploy-release-snapshot step
- create container image tag based on the branch (if it is no release)
- set build-time version based on the branch (if it is no release)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -64,7 +64,14 @@ jobs:
...
           name: Build binary
-          command: go build -o ./build/_output/bin/dynatrace-oneagent-operator ./cmd/manager
+          command: |
+            if [[ -z "$CIRCLE_TAG" ]]; then
+               version="snapshot-${CIRCLE_BRANCH}"
+            else
+               version="${CIRCLE_TAG}"
+            fi
+            echo ${version}
+            go build -ldflags="-X \'github.com/Dynatrace/dynatrace-oneagent-operator/version.Version=${version}\'" -o ./build/_output/bin/dynatrace-oneagent-operator ./cmd/manager
```
Comment (lrgar) [inline .circleci/config.yml]: Can you remove the slashes for the single quotes? It's making it not set the correct field.

```
$ docker run quay.io/dynatrace/dynatrace-oneagent-operator:snapshot-APM-207673-add-tag-per-branch dynatrace-oneagent-operator 2>&1 | grep "Version of dynatrace-oneagent-operator"
{"level":"info","ts":"2019-12-04T12:53:00.695Z","logger":"cmd","msg":"Version of dynatrace-oneagent-operator: snapshot"}
```

**Comment by lrgar** [comment]
Any thoughts on tag sanitization? We're currently using branch names such as "feature/theNextBigThing" or "bugfix/something-broken", and my understanding is that slashes can't be used as tags for example, right?

**Comment by lrgar** [comment]
Oh, and please update the Deployment's image in both `deploy/kubernetes.yaml` and `deploy/openshift.yaml` to use the tag `snapshot-master` since `snapshot` won't be updated anymore.

File: .circleci/config.yml (hunk 1 above)
Comment (thomschuetz) [reply .circleci/config.yml]: done

**Comment by thomschuetz** [comment]
> Any thoughts on tag sanitization? We're currently using branch names such as "feature/theNextBigThing" or "bugfix/something-broken", and my understanding is that slashes can't be used as tags for example, right?

Replaced slashes against dashes  ... should work now

**Comment by thomschuetz** [comment]
> Oh, and please update the Deployment's image in both `deploy/kubernetes.yaml` and `deploy/openshift.yaml` to use the tag `snapshot-master` since `snapshot` won't be updated anymore.

Done

**Pull request merged**